pkg/monitors/jenkins/jobmetrics.go

```diff
@@ -0,0 +1,115 @@
+package jenkins
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/bndr/gojenkins"
+	"github.com/signalfx/golib/v3/datapoint"
+	jc "github.com/signalfx/signalfx-agent/pkg/monitors/jenkins/client"
+)
+
+func (m *Monitor) jobMetrics(jkClient jc.JenkinsClient) ([]*datapoint.Datapoint, error) {
+	// GetAllJobs returns the first level jobs
+	firstLevelJobs, err := jkClient.GoJenkins.GetAllJobs()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get firstLevelJobs %v", err)
+	}
+
+	allJobs := getAllJobs(firstLevelJobs)
```

Comment:
Wonder if we should cache this and only run periodically?

Reply:
Dani and I talked about this before and we decided that it should run everytime to be the most responsive, but only if it was not a huge burden on the Jenkins server.